### PR TITLE
Remove misleading comment

### PR DESCRIPTION
**Pull request by xclerc**
This pull request simply deletes a comment which happens
to be inconsistent with the code in another module.
Precisely, `Tailcall (Self _)` does not raise (See [here](https://github.com/ocaml-flambda/flambda-backend/blob/main/backend/cfg/linear_to_cfg.ml#L212)).

### CODE REVIEW

File: backend/cfg/eliminate_fallthrough_blocks.ml [hunk 1]
```diff
@@ -35,8 +35,6 @@ let is_fallthrough_block cfg_with_layout (block : C.basic_block) =
     || block.is_trap_handler
     || List.length block.body > 0
     || block.can_raise
```
Comment (gretay-js) [inline backend/cfg/eliminate_fallthrough_blocks.ml:37]: It is safe to remove the disjunct `block.can_raise`  that the comment was referring to.

**Review by gretay-js — CHANGES_REQUESTED** [review]
While reviewing this change, I've noticed that `Mach.operation_can_raise` returns `false` for all tailcall operations, but `Linear_to_cfg.can_raise_terminator` returns true for Tailcall (Func). We should probably keep it consistent between Mach and Linear_to_Cfg, although I am not sure which one is correct. 
This is related to recent changes and in particular this comment: https://github.com/ocaml/ocaml/pull/10354/files#r616407258

**Comment by mshinwell** [comment]
Can we try to get this PR fixed and merged?

**Review by gretay-js — APPROVED** [review]

**Comment by mshinwell** [comment]
@xclerc Can we try to get this PR merged please?

**Comment by xclerc** [comment]
I think I have pushed the changes requested by @gretay-js.

**Comment by gretay-js** [comment]
> I think I have pushed the changes requested by @gretay-js.

thanks! it just needs a rebase.

**Pull request merged**